lib/Sema/CodeSynthesis.cpp

```diff
@@ -1310,6 +1310,7 @@ static void synthesizeLazyGetterBody(TypeChecker &TC, AccessorDecl *Get,
                                    Ctx.getIdentifier("tmp2"),
                                    Get);
   Tmp2VD->setType(VD->getType());
+  Tmp2VD->setInterfaceType(VD->getInterfaceType());
```

Comment:
I thought we were done with having to set both of these, no? @slavapestov

Reply:
The rule is you have to set at least the interface type. Calling setType() is optional. If you don’t do it, the first getType() call takes the interface type mapped out of context and caches that. However, I left the setType() method there since in a few places we already have the contextual type handy, so it seemed cleaner than computing it again. We could remove setType() without any ill effects, or add a verifier check that the two types match.